### DISCUSSION TITLE
Adding custom fields

### DISCUSSION THREAD

**Discussion by erkswede** [kimai/kimai#2830] [Questions & Answers] [answered]
Greetings all!

I'm currently trying out Kimai2, self hosted. I need to add a few editable custom fields to my timesheets and read the doc at https://www.kimai.org/documentation/meta-fields.html

I didn't understand however _where_ I need to add the code provided as an example in that doc?
Also, if I make these modifications, will I break the upgradeability of Kimai2, that is, will I need to somehow manage the code I added for my custom fields when upgrading to upcoming releases?

Any help and or pointers appreciated!

**Comment by kevinpapst** [comment, marked as answer]
You can buy the plugin https://www.kimai.org/store/custom-fields-bundle.html and skip doing the work manually.
If you are uncertain about the plugin, there is a Demo where you can try it: https://www.kimai.org/demo/

Or use the DemoBundle https://github.com/Keleo/DemoBundle and checkout how it should be done, as it includes a CustomField as well.

**Reply by erkswede** [reply]
Thanks for the quick reply!

So doing this myself is in fact writing my own plugin?

I'm aware of the Custom Fields plugin and I will most likely end up with that. I'm currently evaluating Kimai2 for usage in charity organizations so I'm trying to cut costs as much as possible.

**Reply by kevinpapst** [reply]
Yes about writing the plugin.
